rpc/client.go

```diff
@@ -187,22 +190,27 @@ func (c *Client) Healthy() <-chan struct{} {
 // Close closes the Closed channel, which triggers the end of the run loop and
 // removal from the clients map.
 func (c *Client) Close() {
+	c.close()
+	<-c.Closed
+}
+
+func (c *Client) close() {
```

Comment:
This is a little bit of a soup, with two close-related channels on the structure and a third passed in on retryOpts.  Could we get a comment or something explaining the various purposes?  Or at least a comment on the internal `close()` method, which ensures that `c.Closer` is closed.